aptos-move/block-executor/src/executor.rs

```diff
@@ -417,46 +405,44 @@ where
         // TODO: should not need to pass base view.
         base_view: &S,
         shared_counter: &AtomicU32,
-        role: CommitRole,
+        shared_commit_state: &ExplicitSyncWrapper<(
+            FeeStatement,
+            Vec<FeeStatement>,
+            Mutex<Option<Error<E::Error>>>,
+        )>,
+        final_results: &ExplicitSyncWrapper<Vec<E::Output>>,
     ) {
         // Make executor for each task. TODO: fast concurrent executor.
         let init_timer = VM_INIT_SECONDS.start_timer();
         let executor = E::init(*executor_arguments);
         drop(init_timer);
 
-        let committing = matches!(role, CommitRole::Coordinator(_));
-
         let _timer = WORK_WITH_TASK_SECONDS.start_timer();
         let mut scheduler_task = SchedulerTask::NoTask;
-        let mut worker_idx = 0;
 
-        let mut accumulated_fee_statement = FeeStatement::zero();
-        let mut txn_fee_statements = Vec::with_capacity(block.len());
         loop {
-            // Only one thread does try_commit to avoid contention.
-            match &role {
-                CommitRole::Coordinator(post_commit_txs) => {
-                    self.coordinator_commit_hook(
-                        self.maybe_block_gas_limit,
-                        scheduler,
-                        post_commit_txs,
-                        &mut worker_idx,
-                        &mut scheduler_task,
+            // Priorotize committing validated transactions
+            while scheduler.should_coordinate_commits() {
+                self.prepare_and_queue_commit_ready_txns(
+                    self.maybe_block_gas_limit,
+                    scheduler,
+                    &mut scheduler_task,
+                    last_input_output,
+                    shared_commit_state,
+                );
+                scheduler.queueing_commits_mark_done();
+            }
+
+            {
```

Comment:
why these brackets? what's scoped here? let's remove if we can, extra indentation doesn't help reviewing, for instance.